src/Stoa.ts

```diff
@@ -381,59 +381,109 @@ class Stoa extends WebService
+     * The parameter `pageSize` is the maximum size that can be obtained
+     *      from one query, default is 10
+     * The parameter `page` is the number on the page, this value begins with 1,
+     *      default is 1
+     * The parameter `type` is the type of transaction to query.
+     *      This can include multiple types.
+     *      Transaction types include "inbound", "outbound", "freeze", "payload".
+     *      The "inbound" is an increased transaction of funds at the address.
+     *      The "outbound" is a transaction with reduced funds at the address.
+     *      Users can select only "inbound", "outbound".
+     *      The "freeze", "payload" are always included.
+     *      default is "inbound,outbound,freeze,payload"
+     * The parameter `beginDate` is the start date of the range of dates to look up.
+     * The parameter `endDate` is the end date of the range of dates to look up.
+     * The parameter `peer` is used when users want to look up only specific
+     *      address of their counterparts.
+     *      Peer is the withdrawal address in the inbound transaction and
+     *      a deposit address in the outbound transaction
      * Returns a set of transactions history of the addresses.
+     * ```
      */
     private getWalletTransactionsHistory (req: express.Request, res: express.Response)
     {
-        let params_addresses: string = String(req.params.addresses);
+        let address: string = String(req.params.address);
 
-        logger.http(`GET /wallet/transactions/history/${params_addresses}}`);
+        logger.http(`GET /wallet/transactions/history/${address}}`);
 
-        let page_size = (req.query.pageSize !== undefined)
-            ? Number(req.query.pageSize.toString())
-            : 10;
+        if ((req.query.beginDate !== undefined) &&
+            !Utils.isPositiveInteger(req.query.beginDate.toString()))
+        {
+            res.status(400).send(`Invalid value for parameter 'beginDate': ${req.query.beginDate.toString()}`);
+            return;
+        }
 
-        let page = (req.query.page !== undefined)
-            ? Number(req.query.page.toString())
-            : 1;
+        if ((req.query.endDate !== undefined) &&
+            !Utils.isPositiveInteger(req.query.endDate.toString()))
+        {
+            res.status(400).send(`Invalid value for parameter 'endDate': ${req.query.endDate.toString()}`);
+            return;
+        }
 
-        if (page_size <= 0)
+        if ((req.query.pageSize !== undefined) &&
+            !Utils.isPositiveInteger(req.query.pageSize.toString()))
         {
-            res.status(400).send(`Positive pageSize expected: ${page_size}`);
+            res.status(400).send(`Invalid value for parameter 'pageSize': ${req.query.pageSize.toString()}`);
             return;
         }
 
-        if (page_size > 30)
+        if ((req.query.page !== undefined) &&
+            !Utils.isPositiveInteger(req.query.page.toString()))
```

Comment:
This is allowed if the page size is larger than the page size if entered.

Reply:
I don't understand your question, so I'd appreciate it if you could write it more specifically.

Reply:
There is a typo. Page cannot be larger than the page size.